moto/glue/glue_schema_registry_utils.py

```diff
@@ -0,0 +1,273 @@
+import re
+import json
+
+from .glue_schema_registry_constants import (
+    MAX_REGISTRY_NAME_LENGTH,
+    RESOURCE_NAME_PATTERN,
+    MAX_ARN_LENGTH,
+    ARN_PATTERN,
+    MAX_DESCRIPTION_LENGTH,
+    DESCRIPTION_PATTERN,
+    MAX_SCHEMA_NAME_LENGTH,
+    DEFAULT_REGISTRY_NAME,
+    REGISTRY_NAME,
+    REGISTRY_ARN,
+    SCHEMA_NAME,
+    SCHEMA_ARN,
+    MAX_TAGS_ALLOWED,
+    MAX_SCHEMA_DEFINITION_LENGTH,
+    SCHEMA_DEFINITION,
+    MAX_SCHEMAS_ALLOWED,
+    MAX_SCHEMA_VERSIONS_ALLOWED,
+    MAX_REGISTRIES_ALLOWED,
+)
+
+from .exceptions import (
+    ResourceNameTooLongException,
+    ParamValueContainsInvalidCharactersException,
+    InvalidSchemaDefinitionException,
+    InvalidRegistryIdBothParamsProvidedException,
+    GSREntityNotFoundException,
+    InvalidSchemaIdBothParamsProvidedException,
+    InvalidSchemaIdInsufficientParamsProvidedException,
+    SchemaNotFoundException,
+    InvalidDataFormatException,
+    InvalidCompatibilityException,
+    InvalidNumberOfTagsException,
+    GSRAlreadyExistsException,
+    ResourceNumberLimitExceededException,
+    DisabledCompatibilityVersioningException,
+)
+
+
+def validate_registry_name_pattern_and_length(param_value):
+    param_name = "registryName"
+    max_name_length = MAX_REGISTRY_NAME_LENGTH
+    pattern = RESOURCE_NAME_PATTERN
+    validate_param_pattern_and_length(param_value, param_name, max_name_length, pattern)
+
+
+def validate_arn_pattern_and_length(param_value):
+    param_name = "registryArn"
+    max_name_length = MAX_ARN_LENGTH
+    pattern = ARN_PATTERN
+    validate_param_pattern_and_length(param_value, param_name, max_name_length, pattern)
+
+
+def validate_description_pattern_and_length(param_value):
+    param_name = "description"
+    max_name_length = MAX_DESCRIPTION_LENGTH
+    pattern = DESCRIPTION_PATTERN
+    validate_param_pattern_and_length(param_value, param_name, max_name_length, pattern)
+
+
+def validate_schema_name_pattern_and_length(param_value):
+    param_name = "schemaName"
+    max_name_length = MAX_SCHEMA_NAME_LENGTH
+    pattern = RESOURCE_NAME_PATTERN
+    validate_param_pattern_and_length(param_value, param_name, max_name_length, pattern)
+
+
+def validate_param_pattern_and_length(
+    param_value, param_name, max_name_length, pattern
+):
+    if len(param_value.encode("utf-8")) > max_name_length:
+        raise ResourceNameTooLongException(param_name)
+
+    if re.match(pattern, param_value) is None:
+        raise ParamValueContainsInvalidCharactersException(param_name)
+
+
+def validate_number_of_tags(tags):
+    if len(tags) > MAX_TAGS_ALLOWED:
+        raise InvalidNumberOfTagsException()
+
+
+def validate_avro_json_schema_definition(data_format, schema_definition):
+    if data_format in ["AVRO", "JSON"]:
+        try:
+            json.loads(schema_definition)
+        except ValueError as err:
+            raise InvalidSchemaDefinitionException(data_format, err)
+
+
+def compare_json_helper(item):
```

Comment:
I recommend comparing the dicts of loaded JSON instead of writing this manually.